runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/source/unbounded/FlinkUnboundedSourceReader.java

```diff
@@ -126,6 +126,9 @@ public InputStatus pollNext(ReaderOutput<WindowedValue<ValueWithRecordId<T>>> ou
     if (reader != null) {
       emitRecord(reader, output);
       return InputStatus.MORE_AVAILABLE;
+    } else if (noMoreSplits()) {
```

Comment:
Just to confirm, does noMoreSplits()=true means there won't be split forever (not just currently)

From https://nightlies.apache.org/flink/flink-docs-master/api/java/org/apache/flink/api/connector/source/SourceReader.html#notifyNoMoreSplits-- iiuc it is this case

Reply:
That's my understanding yes.